radix-engine/src/blueprints/resource/resource_manager_common.rs

```diff
@@ -225,15 +194,26 @@ pub fn globalize_non_fungible_with_initial_supply<Y>(
     owner_role: OwnerRole,
     object_id: NodeId,
     resource_address_reservation: GlobalAddressReservation,
-    access_rules: BTreeMap<ResourceAction, (AccessRule, AccessRule)>,
+    mut main_roles: RolesInit,
```

Comment:
The way we converts `NonFungibleResourceFeatures` into features and roles, and then `verify_and_defaults_to_main_roles`, seems unnecessary?

One of the benefits of using `NonFungibleResourceFeatures` to have structured data and reduce validation needed.

Reply:
You're right, the old `verify_and_defaults_to_main_roles` should have been removed.